SCClassLibrary/Common/GUI/tools/Stethoscope.sc

```diff
@@ -393,7 +397,22 @@ Stethoscope {
 
 	size_ { arg value;
 		var sz = value.asSize;
-		if( window.notNil ) { window.setInnerExtent(sz.width,sz.height) };
+		if(window.notNil) {
+			var widthIsWithinBounds = sz.width + window.bounds.left < Window.screenBounds.width;
+			var heightIsWithinBounds = window.bounds.bottom - sz.height > 0;
+			if(widthIsWithinBounds && heightIsWithinBounds) {
+				window.setInnerExtent(sz.width,sz.height);
+			} {
+				window.bounds_(
+					Rect(
+						if(widthIsWithinBounds) { window.bounds.left } { Window.screenBounds.width - sz.width },
```

Comment:
Maybe check if this may become negative?
- `Window.screenBounds.width - sz.width`
- `window.bounds.bottom - sz.height`
They probably shouldn't, right?

What you do here could be a method in view, which sets the view size in a particular way. If you want to spend the time thinking this through in general, it may be used in other places as well.

Generally speaking, a lot of the GUI bounds suggestions may point toward a more general solution?

Reply:
>`Window.screenBounds.width - sz.width`
>`window.bounds.bottom - sz.height`

They do not become negative.

The following three screenshots summarise what happens when you press m:

- The main display is on top. 
  A and B appear as C 
 ![Screenshot 2024-07-07 at 19 42 47](https://github.com/supercollider/supercollider/assets/416281/650e9874-7fbe-449f-8083-654357e1a867)

-  The main display is in the centre.
  A -> B, C -> D, E -> F:
![Screenshot 2024-07-07 at 21 38 20](https://github.com/supercollider/supercollider/assets/416281/97c993c2-49b0-48c5-aad4-9b63c02effab)

-  Main display is at the bottom
  A -> B, C -> D, E -> F:
  ![Screenshot 2024-07-07 at 19 46 04](https://github.com/supercollider/supercollider/assets/416281/2f85c6c3-563e-4fbf-862a-eb856df537b1)